exporter/collector/metricsexporter.go

```diff
@@ -97,23 +123,82 @@ func (me *metricsExporter) pushMetrics(ctx context.Context, m pdata.Metrics) err
 			mes := ilm.Metrics()
 			for k := 0; k < mes.Len(); k++ {
 				metric := mes.At(k)
+				for _, md := range me.mapper.metricDescriptor(metric) {
+					me.mds <- md
+				}
 				timeSeries = append(timeSeries, me.mapper.metricToTimeSeries(monitoredResource, extraLabels, metric)...)
 			}
 		}
 	}
 
 	// TODO: self observability
+	// TODO: Figure out how to configure service time series calls.
+	if false {
+		err := me.createServiceTimeSeries(ctx, timeSeries)
+		recordPointCount(ctx, len(timeSeries), m.DataPointCount()-len(timeSeries), err)
+		return err
+	}
 	err := me.createTimeSeries(ctx, timeSeries)
 	recordPointCount(ctx, len(timeSeries), m.DataPointCount()-len(timeSeries), err)
 	return err
 }
 
+// Reads metric descriptors from the md channel, and reports them (once) to GCM.
+func (me *metricsExporter) exportMetricDescriptorRunner() {
+	mdCache := make(map[string]*metricpb.MetricDescriptor)
+	// We iterate over all metric descritpors until the channel is closed.
+	// Note: if we get terminated, this will still attempt to export all descriptors
+	// prior to shutdown.
+	for md := range me.mds {
+		// Not yet sent, now we sent it.
+		// TODO - check to see if this is a service/system metric and doesn't send descriptors.
+		if !me.cfg.MetricConfig.SkipCreateMetricDescriptor && md != nil && mdCache[md.Type] == nil {
+			err := me.exportMetricDescriptor(context.Background(), md)
```

Comment:
You probably want a timeout for this ctx, or is this handled automatically by the client library?

Reply:
I believe we should be able to configure that on gRPC clients and it'll attach to an existing timeout if one exists, or create one.   However, I'm rather weak at Go, as you know :)

Reply:
I wasn't sure so looked around: https://pkg.go.dev/cloud.google.com/go#hdr-Timeouts_and_Cancellation TLDR; the client lib will set a default timeout if the context doesn't have one already.

@dashpole probably knows better, is the default reasonable for this or should we hardcode something?

Reply:
I think there is a timeout in the googlecloud exporter's config.  I'd expect that to apply to the CMD call as well.  Otherwise, i'd expect it to use the default timeout

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ebea5bc82e5a146db11d4b496a7a62e2151fe89a/exporter/googlecloudexporter/googlecloud.go#L141

Reply:
Discussed in meeting, would be better to send the context in the channel from the other goroutine.